File: chapters/src/main/java/cn/rubintry/chapters/viewmodel/SubscriptionViewModel.java
package cn.rubintry.chapters.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

import cn.rubintry.chapters.model.SubscriptionModel;
import cn.rubintry.common.model.BaseModel;
import cn.rubintry.common.utils.http.OkHttpUtils;
import cn.rubintry.common.utils.http.ResponseCallback;

/**
 * @author logcat
 */
public class SubscriptionViewModel extends ViewModel {
    private MutableLiveData<BaseModel<List<SubscriptionModel>>> subscriptionData;

    public MutableLiveData<BaseModel<List<SubscriptionModel>>> getSubscriptionData(){
        if(subscriptionData == null){
            subscriptionData = new MutableLiveData<>();
        }
        loadData();
        return subscriptionData;
    }

    private void loadData() {
        OkHttpUtils.getInstance()
                .get("/wxarticle/chapters/json")
                .request(new ResponseCallback<BaseModel<List<SubscriptionModel>>>() {
                             @Override
                             protected void onSuccess(BaseModel<List<SubscriptionModel>> listBaseModel) {
                                 subscriptionData.postValue(listBaseModel);
                             }

                             @Override
                             protected void onFailed(Exception e, BaseModel<List<SubscriptionModel>> listBaseModel) {
                                 subscriptionData.postValue(listBaseModel);
                             }
                         }

                );
    }
}

File: chapters/src/main/java/cn/rubintry/chapters/viewmodel/SubscriptionViewModel.kt
package cn.rubintry.chapters.viewmodel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import cn.rubintry.chapters.model.SubscriptionModel
import cn.rubintry.common.model.BaseModel
import cn.rubintry.common.utils.http.OkHttpUtils
import cn.rubintry.common.utils.http.ResponseCallback

/**
 * @author logcat
 */
class SubscriptionViewModel : ViewModel() {
    private var subscriptionData: MutableLiveData<BaseModel<List<SubscriptionModel>>> = MutableLiveData()

    fun getSubscriptionData(): MutableLiveData<BaseModel<List<SubscriptionModel>>> {
        loadData()
        return subscriptionData
    }

    fun loadData() {

        OkHttpUtils.instance
            ?.get("/wxarticle/chapters/json")
            ?.request(object  : ResponseCallback<BaseModel<List<SubscriptionModel>>>(){
                override fun onSuccess(t: BaseModel<List<SubscriptionModel>>) {
                    subscriptionData.postValue(t)
                }

                override fun onFailed(e: Exception?, t: BaseModel<List<SubscriptionModel>>?) {
                    subscriptionData.postValue(t)
                }

            })
    }
}